packages/rollup-contracts/contracts/state-tree/FullPatriciaTree.sol

```diff
@@ -0,0 +1,380 @@
+    function verifyEdgeInclusionProof(
+        bytes32 rootHash,
+        bytes32 edgeCommittment,
+        D.Label memory fullEdgeLabel,
+        uint branchMask,
+        bytes32[] memory siblings
+    ) internal pure {
+        // We will progressively "eat" into the label from the right as we hash up to the root.
+        D.Label memory remaining = fullEdgeLabel;
+        // We will progressively hash the current edge up with its siblings until we get the root edge.
+        D.Edge memory currentEdge;
+        // To start, this is the edge we are verifying so it's the edgeHash which was input
+        currentEdge.node = edgeCommittment;
+        // Iterate over each set bit in the branch mask to build parent edges, starting from the right.
+        for (uint i = 0; branchMask != 0; i++) {
+            // Find the lowest index nonzero bit in the mask, where rightmost == index 0
+            uint bitSet = Utils.lowestBitSet(branchMask);
+            // Remove from bitmask as we are about to process it
+            branchMask &= ~(uint(1) << bitSet);
+            // The label for the current edge is the suffix of the remaining label proceeeding the set bit
+            (remaining, currentEdge.label) = Utils.splitAt(remaining, 255 - bitSet); // (255 - bitSet) since bitset indexes from the right
+            // The bitSet'th bit in the key determines whether the sibling is left or right.
+            uint bit;
+            // chop this bit off the label, it is implicit in the merkle path so will not be included in a label
+            (bit, currentEdge.label) = Utils.chopFirstBit(currentEdge.label);
+            bytes32[2] memory edgeHashes;
+            edgeHashes[bit] = edgeHash(currentEdge);
+            edgeHashes[1 - bit] = siblings[siblings.length - i - 1];
+            currentEdge.node = keccak256(abi.encode(edgeHashes[0], edgeHashes[1]));
+        }
+        // no more branching, so the remaining label is the root edge's label
+        currentEdge.label = remaining;
+        require(rootHash == edgeHash(currentEdge), 'Edge inclusion proof verification failed: root hashes do not match.');
+    }
+
+    function verifyProof(
+        bytes32 rootHash,
+        bytes32 key,
+        bytes memory value,
+        uint branchMask,
+        bytes32[] memory siblings
+    ) public pure {
+        // The edge above a leaf commits to the leaf value (i.e. what was actually set) 
+        bytes32 edgeCommittment = keccak256(value);
+        // The full "label" for a leaf node is the entirety of the key.
+        D.Label memory fullLabel = D.Label(key, 256);
+
+        verifyEdgeInclusionProof(
+            rootHash,
+            edgeCommittment,
+            fullLabel,
+            branchMask,
+            siblings
+        );
+    }
+
+    function verifyNonInclusionProof(
+        bytes32 rootHash,
+        bytes32 key,
+        bytes32 conflictingEdgeFullLabelData,
+        uint conflictingEdgeFullLabelLength,
```

Comment:
Any reason that `full` is used here and in other variables in this function? Can we assume it is full unless otherwise specified?